### PR TITLE
[Imporvement]: Add spotless:validate to ams/api and ams/optimizer modules

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

This is a subtask of https://github.com/NetEase/amoro/issues/1402. To replace checkstyle to spotless as default codestyle formatter.

## Brief change log

- Enable spotless configurations for ams/api and ams/optimizer modules.
- Skip maven-checkstyle-plugin for ams/api and ams/optimizer modules.
- Run spotless:apply for ams/api and ams/optimizer modules.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

### CODE REVIEW

**Review by zhongqishang — APPROVED** [review]
LGTM.

File: ams/api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java [hunk 1]
```diff
@@ -223,13 +237,23 @@ public String serverUrl() {
...
   public String toString() {
-    return "ArcticThriftUrl{" +
-        "schema='" + schema + '\'' +
-        ", host='" + host + '\'' +
-        ", port=" + port +
-        ", catalogName='" + catalogName + '\'' +
-        ", socketTimeout=" + socketTimeout +
-        ", url='" + url + '\'' +
-        '}';
+    return "ArcticThriftUrl{"
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java:240]: How about changing the implementation with `MoreObjects`?

**Review by wangtaohz — APPROVED** [review]
LGTM

File: ams/api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java (hunk 1 above)
Comment (baiyangtx) [reply ams/api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java:240]: This could be in an another PR.

Comment (baiyangtx) [reply ams/api/src/main/java/com/netease/arctic/ams/api/client/ArcticThriftUrl.java]: fixed.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**